Update base AMI to Debian Bookworm

jsf9k opened the pull request:
## 🗣 Description ##

This pull request updates the base AMI from Debian Bullseye to Debian Bookworm.

## 💭 Motivation and context ##

Debian Bookworm was officially released on June 10, 2023.

## 🧪 Testing ##

All automated tests pass.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] *All* future TODOs are captured in issues, which are referenced in code comments.
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

## ✅ Pre-merge checklist ##

<!-- Remove any of the following that do not apply. -->
<!-- These boxes should remain unchecked until the pull request has been -->
<!-- approved. -->

- [x] Finalize version.

## ✅ Post-merge checklist ##

<!-- Remove any of the following that do not apply. -->

- [ ] Create a release.

jsf9k committed:
Update base AMI to Debian Bookworm

Debian Bookworm was officially released on June 10,2023.

mcdonnnj reviewed (approved):
![image](https://github.com/cisagov/skeleton-packer/assets/50747025/5fc8825b-d7a7-4f95-9474-131e19757b29)

mcdonnnj commented:
@jsf9k Until we come to a final determination on handling the change in Bookworm from using [resolvconf](https://packages.debian.org/bookworm/resolvconf) to [systemd-resolved](https://packages.debian.org/bookworm/systemd-resolved) for DNS resolution would it be fair to mark this as blocked?

jsf9k committed:
Add a playbook to install prerequisites for fixing a broken Netplan configuration

See these issues for more information:
- cisagov/skeleton-packer#300
- canonical/cloud-init#4764

jsf9k commented:
Please note that the linter failures are because this PR is lacking the changes from PR #295.  Once that PR is merged then this one should also pass its checks.

dav3r reviewed (approved):
👍 👍

mcdonnnj commented on src/fix-netplan-configuration.yml:
Thoughts on changing the name of this playbook? It reads like it fixes the Netplan configuration when it's installing prerequisites for the downstream bits that fix the configuration.

dav3r replied:
That is a good point - it confused me at first as well.  How about `install-prereqs-for-netplan-config-fix.yml`?  Do any of you have something more concise?

jsf9k replied:
Please enjoy commit f0f49a1.

jsf9k committed:
Rename playbook to better describe what it does

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>
Co-authored-by: dav3r <david.redmin@gwe.cisa.dhs.gov>

jsf9k commented:
@mcdonnnj - Can you post something here and/or in #300 when your CyHy testing is complete?

jsf9k committed:
Add comments mentioning cisagov/skeleton-packer#301

These comments remind us to remove the code to fix up the Netplan
configuration generated by cloud-init when and if that becomes
possible.

jsf9k commented:
> @mcdonnnj - Can you post something here and/or in #300 when your CyHy testing is complete?

@mcdonnnj - Does cisagov/cyhy_amis#746 mean that your testing is complete and successful?

mcdonnnj commented:
> > @mcdonnnj - Can you post something here and/or in #300 when your CyHy testing is complete?
> 
> @mcdonnnj - Does [cisagov/cyhy_amis#746](https://github.com/cisagov/cyhy_amis/pull/746) mean that your testing is complete and successful?

It does now. I meant to create that as a draft PR while I did some last verification. I have finished that verification and added a small update to #300 confirming success.

mcdonnnj committed:
Finalize version from 1.0.0-rc.2 to 1.0.0

mcdonnnj committed:
Reorder YAML keys to placate ansible-lint
